src/dex/uniswap-v2/uniswap-v2.ts

```diff
@@ -81,44 +81,38 @@ export class UniswapV2EventPool extends StatefulEventSubscriber<UniswapV2PoolSta
   decoder = (log: Log) => this.iface.parseLog(log);
 
   constructor(
-    protected parentName: string,
     protected dexHelper: IDexHelper,
+    parentName: string,
     private poolAddress: Address,
-    private token0: Token,
-    private token1: Token,
+    private suffix: string,
     // feeCode is ignored if DynamicFees is set to true
     private feeCode: number,
-    logger: Logger,
     private dynamicFees = false,
+    logger: Logger,
     // feesMultiCallData is only used if dynamicFees is set to true
     private feesMultiCallEntry?: { target: Address; callData: string },
     private feesMultiCallDecoder?: (values: any[]) => number,
     private iface: Interface = uniswapV2Iface,
   ) {
-    super(
-      parentName +
-        ' ' +
-        (token0.symbol || token0.address) +
-        '-' +
-        (token1.symbol || token1.address) +
-        ' pool',
-      logger,
-    );
+    super(dexHelper, parentName, `${poolAddress}${suffix}`, logger);
+    this.addressesSubscribed.push(poolAddress);
   }
 
   protected processLog(
     state: DeepReadonly<UniswapV2PoolState>,
     log: Readonly<Log>,
   ): AsyncOrSync<DeepReadonly<UniswapV2PoolState> | null> {
-    const event = this.decoder(log);
-    switch (event.name) {
-      case 'Sync':
-        return {
-          reserves0: event.args.reserve0.toString(),
-          reserves1: event.args.reserve1.toString(),
-          feeCode: state.feeCode,
-        };
-    }
+    try {
+      const event = this.decoder(log);
+      switch (event.name) {
+        case 'Sync':
+          return {
+            reserves0: event.args.reserve0.toString(),
+            reserves1: event.args.reserve1.toString(),
+            feeCode: state.feeCode,
+          };
+      }
+    } catch (e) {}
```

Comment:
I don't think it is a good idea, to silently skip in case if unexpected error. Maybe better to add more granular approach? Silent the error we expect and log if something more happens?